plugins/processors/script/tengo_script.go

```diff
@@ -0,0 +1,120 @@
+package script
+
+import (
+	"context"
+	_ "embed" // used to print the embedded assets
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/d5/tengo/v2"
+	"github.com/odpf/meteor/models"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/plugins/internal/tengoutil"
+	"github.com/odpf/meteor/plugins/internal/tengoutil/structmap"
+	"github.com/odpf/meteor/registry"
+	"github.com/odpf/salt/log"
+)
+
+func init() {
+	if err := registry.Processors.Register("script", func() plugins.Processor {
+		return New(plugins.GetLog())
+	}); err != nil {
+		return
+	}
+}
+
+//go:embed README.md
+var summary string
+
+type Config struct {
+	Engine string `mapstructure:"engine" validate:"required,oneof=tengo"`
+	Script string `mapstructure:"script" validate:"required"`
+}
+
+// Processor executes the configured Tengo script to transform the given asset
+// record.
+type Processor struct {
+	plugins.BasePlugin
+	config Config
+	logger log.Logger
+
+	compiled *tengo.Compiled
+}
+
+var sampleConfig = heredoc.Doc(`
+	engine: tengo
+	script: |
+	  asset.owners = append(asset.owners || [], { name: "Big Mom", email: "big.mom@wholecakeisland.com" })
+`)
+
+var info = plugins.Info{
+	Description:  "Transform the extracted asset with the configured Tengo script",
+	SampleConfig: sampleConfig,
+	Summary:      summary,
+	Tags:         []string{"processor", "transform", "script"},
+}
+
+// New create a new processor
+func New(logger log.Logger) *Processor {
+	p := &Processor{
+		logger: logger,
+	}
+	p.BasePlugin = plugins.NewBasePlugin(info, &p.config)
+
+	return p
+}
+
+func (p *Processor) Init(ctx context.Context, config plugins.Config) error {
+	if err := p.BasePlugin.Init(ctx, config); err != nil {
+		return fmt.Errorf("script processor init: %w", err)
+	}
+
+	s := tengoutil.NewSecureScript(([]byte)(p.config.Script))
```

Comment:
Just an idea, wdyt if we prepend the script with helper functions like below.

```
text := import("text")

merge := func(m1, m2) {
    for k, v in m2 {
        m1[k] = v
    }
    return m1
}
```

CMIIW, adding this should help users from defining their own `merge` function on every recipe.

Reply:
There is explicit support for defining an importable module but at this point, considering we don't have a clear idea of what all helpers could be useful, I would lean towards gathering some experience before introducing the helper module.

Reply:
Ok so the idea is, if we are going to use this `script` processor to replace our `labels` processor in all of our recipes we would need to define this `merge` everytime right?

Was thinking if we have that, users can easily migrate to `script` processor. Or we can also define `set_label` function if that is more helpful.

Reply:
IMO, using the script processor for setting labels is overkill and it will not nearly be as clear as using the labels processor. I think it still makes sense to provide processors for common use cases that are easier to define, test and maintain from the user's perspective.

And regarding `merge`, I am not sure if we should provide that as a standard way for setting a value into the map. It _might_ be better to provide a `set_map` helper that would initialise the map if required. Might be worth analysing in a different PR.